templates/admin/src/index.tsx.ts

```diff
@@ -1,12 +1,41 @@
-import { readFile, TemplateFunction } from "../../../src/lib/createAdapter";
+import { TemplateFunction } from "../../../src/lib/createAdapter";
 
 const templateFunction: TemplateFunction = answers => {
 
 	const useTypeScript = answers.language === "TypeScript";
 	const useReact = answers.adminReact === "yes";
 	if (!(useTypeScript && useReact)) return;
+	
+	const template = `
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import { MuiThemeProvider } from "@material-ui/core/styles";
+import theme from "@iobroker/adapter-react/Theme";
 
-	return readFile("index.raw.tsx", __dirname);
+import Utils from "@iobroker/adapter-react/Components/Utils";
+
+import App from "./app";
+
+window["adapterName"] = ${JSON.stringify(answers.adapterName)};
+let themeName = Utils.getThemeName();
+
+function build(): void {
```

Comment:
Is the build function used anywhere else than here? It doesn't seem necessary to wrap the render call.

Reply:
Yes, about 6 lines later ;-). It calls itself when the Theme changes. Honestly I don't understand the entire Theme changing yet, as also ESlint complains about the `onThemeChange` property missing.

Reply:
Yeah, but you could just not make it a separate function and call ReactDOM.render(...) directly.

Reply:
So, what's the conclusion?

Reply:
I'm not sure, I understood your conclusion. The `build()` method is called in two places:
- in the root of `index.tsx` (L34)
- in the `onThemeChange` handler (L26)

See here: https://github.com/UncleSamSwiss/create-adapter/blob/a92a494c813d94710c6208f16d3995fb95cc8145/templates/admin/src/index.tsx.ts#L20-L34

Reply:
My bad... I checked "get all references" in VSCode and did not see the self-reference